zarrs/src/node/node_name.rs

```diff
@@ -35,7 +35,9 @@ impl NodeName {
     /// `name` is not validated, so this can result in an invalid node name.
     #[must_use]
     pub unsafe fn new_unchecked(name: impl Into<String>) -> Self {
-        Self(name.into())
+        let name: String = name.into();
+        debug_assert!(Self::validate(&name));
+        Self(name)
```

Comment:
While tracking down this bug I thought we were creating an invalid NodeName here -- we weren't but this seemed like a generally good change, but happy to revert or break out if you prefer.

Reply:
Nice one! Adding these in unchecked methods is always good.